articles/rules/redirect.md

```diff
@@ -32,18 +31,22 @@ function (user, context, callback) {
 
 Once all rules have finished executing, the user will be redirected to the specified URL.
 
-## What to do after redirecting
+## What to do afterwards
 
 An authentication transaction that has been interrupted by setting `context.redirect` can be resumed by redirecting the user to the following URL:
 
 ```text
 https://${account.namespace}/continue?state=THE_ORIGINAL_STATE
 ```
 
+::: panel-info State
+State is an opaque value the clients adds to the initial request that Auth0 includes when redirecting back to the client. We highly recommend using this param, since it can be used by the client to prevent [CSRF attacks](/security/common-threats#cross-site-request-forgery-xsrf-or-csrf-). By `THE_ORIGINAL_STATE` we mean the value you provided during user authentication.
```

Comment:
That's not right. In this context, `THE_ORIGINAL_STATE` is not the state the client passes to Auth0 in the original request. `THE_ORIGINAL_STATE` is the state that Auth0 generates and sends to the redirect URL.

If your Rule has this configured:
```
function (user, context, callback) {
    context.redirect = {
        url: "https://redirect.com/foo"
    };
    return callback(null, user, context);
}
```

For example.

1. `client.com/login`
2. `<account>.auth0.com/authorize?...&state=foo123`
3. Social IdP / DB connection call
4. `redirect.com/foo?state=bar42` (note that this is entirely independent of the state the client passes to Auth0 - this state is generated for the transaction between Auth0 and the redirect URL)
5. `<account>.auth0.com/continue?state=bar42`
6. Rules run again with `context.protocol === "redirect-callback"`
7. `client.com/callback?code=...&state=foo123`

In this context, `redirect.com/foo` needs to handle the `state` parameter `bar42` and send it back to Auth0's `/continue`.